pandas/core/groupby/generic.py

```diff
@@ -615,7 +615,11 @@ def _transform_fast(self, result, obj, func_nm):
         ids, _, ngroup = self.grouper.group_info
         output = []
         for i, _ in enumerate(result.columns):
-            res = algorithms.take_1d(result.iloc[:, i].values, ids)
+            if func_nm in base.reduction_kernels:
+                # only broadcast results if we performed a reduction
+                res = algorithms.take_1d(result.iloc[:, i].values, ids)
```

Comment:
would ._values here work instead of .values?  generally less lossy

Reply:
Interestingly, the SeriesGroupby equivalent already does. Changed.